### PR TITLE
Don't fetch blobs in trees/latest if not the first call

### PR DESCRIPTION
**Pull request by jatgarg**
## Description

Don't fetch blobs in trees/latest if not the first call as they are not required. Just the trees are required by summarizer.

### CODE REVIEW

File: packages/drivers/odsp-driver/src/fetchSnapshot.ts [hunk 1]
```diff
@@ -67,9 +67,9 @@ export async function fetchSnapshot(
 
     if (fetchFullSnapshot) {
         if (versionId !== "latest") {
-            queryParams = { channels: 1, blobs: 2 };
+            queryParams = { blobs: 2 };
         } else {
-            queryParams = { deltas: 1, channels: 1, blobs: 2 };
+            queryParams = { deltas: 1, blobs: 2 };
```
Comment (vladsud) [inline packages/drivers/odsp-driver/src/fetchSnapshot.ts:72]: I'm a bit confused on the flow. 
1. This is only used through getSnapshotTree() call, and fetchFullSnapshot is always true (in production code). And only if we do not have tree in cache. I'd assume we should always download just a tree, i.e. fetchFullSnapshot should be always false, no?
2. It feels like the most important stuff we need is channels, does dropping them means - we use default (they are included), or not?

Comment (jatgarg) [reply packages/drivers/odsp-driver/src/fetchSnapshot.ts:72]: 1.) I am not aware why fetchFullSnapshot was added and why it is set to true in the production. Just wanted to keep the behavior as is. However, this does not do anything as by default they supply both blobs and deltas. So I am going to remove them.
2.) I confirmed that Channels is not even processed by the server. Only deltas and blobs. So, they always supply trees by default. That is why I am removing them here.

File: packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts [hunk 2]
```diff
@@ -308,11 +308,17 @@ export class OdspDocumentStorageService extends OdspDocumentStorageServiceBase {
...
-                            if (retrievedSnapshot === undefined) {
-                                retrievedSnapshot = await this.fetchSnapshot(hostSnapshotOptions, scenarioName);
+                            if (retrievedSnapshot !== undefined) {
+                                method = "cache";
+                            } else {
+                                method = "network";
+                                const options: ISnapshotOptions = { ...hostSnapshotOptions };
+                                // Don't fetch the blobs/deltas if it is not the first call.
+                                if (!this.firstVersionCall) {
+                                    options.blobs = 0;
```
Comment (vladsud) [inline packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts:318]: Do we want to have options.deltas = 0? Or is this a default? I think we need to put a comment somewhere telling what are the defaults, as I have no clue :)

File: packages/drivers/odsp-driver/src/fetchSnapshot.ts [hunk 3]
```diff
@@ -63,19 +63,8 @@ export async function fetchSnapshot(
     snapshotDownloader: (url: string, fetchOptions: { [index: string]: any; }) => Promise<IOdspResponse<unknown>>,
 ): Promise<ISnapshotContents> {
     const path = `/trees/${versionId}`;
-    let queryParams: ISnapshotOptions = {};
-
-    if (fetchFullSnapshot) {
```
Comment (vladsud) [inline packages/drivers/odsp-driver/src/fetchSnapshot.ts:68]: fetchFullSnapshot arg is no longer used, so it should be removed, right?
That said, does it mean that we always fetch full snapshot? Or only tree? What are the default behavior for ODSP?
I think this API is used to fetch only tree, so we should ensure (trough defaults, or through explicit args) it does that.
I'd chose being explicit, i.e. our code not depend on ODSP defaults and always clearly communicating what it wants (i.e. setting all these queryParams)

**Review by vladsud — APPROVED** [review]
Please validate in debugger (using examples) that we are not loading all blobs one by one, i.e. we either fetch full snapshot, or fetching a tree (and never fetch blobs from it)

**Comment by jatgarg** [comment]
Seems to work fine as I checked in debugger.

**Pull request merged**